test/distributed/_tensor/test_xla_integration.py

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+# Owner(s): ["oncall: distributed"]
+
+import os
+import unittest
+from functools import wraps
+from typing import Any, Callable, Dict, Tuple
+
+import torch
+from torch.distributed._tensor import DeviceMesh, distribute_tensor, Shard
+from torch.testing._internal.common_utils import run_tests, TestCase
+
+
+# wrapper to check xla test requirements
+def with_xla(func: Callable) -> Callable:
+    assert func is not None
+
+    @wraps(func)  # pyre-ignore[6]
+    def wrapper(
+        self, *args: Tuple[object], **kwargs: Dict[str, Any]  # type: ignore[misc]
+    ) -> None:
+        # TODO(yeounoh) replace this with xr.use_spmd() when we deprecate the flag.
+        os.environ["XLA_USE_SPMD"] = "1"
+        try:
+            import torch_xla  # type:ignore[import]  # noqa: F401
+        except ImportError as exc:
+            raise unittest.SkipTest("torch_xla is not installed.") from exc
+        self.device_type = "xla"
+        func(self, *args, **kwargs)  # type: ignore[misc]
+        os.environ["XLA_USE_SPMD"] = "0"
+
+    return wrapper
+
+
+class DTensorXLAIntegrationTest(TestCase):
+    @property
+    def device_count(self) -> int:
```

Comment:
do you need to check the device count < 4 and skip the test too?

Reply:
Let's use the actual local device count instead, I've updated the test cases.